### ISSUE TITLE
make tests for the tutorials

### ISSUE THREAD

**Issue by goodfeli** [tensorflow/cleverhans#67]

**Comment by goodfeli** [comment]
The existing tests use code duplication rather than calling the main() function of the tutorial, so they need to be independently maintained

**Comment by npapernot** [comment]
@goodfeli do you consider this closed since we put together the Travis conf in #100?

In my opinion, we could probably use more (~unit) tests

**Comment by goodfeli** [comment]
No, this isn't closed. A way of rephrasing this issue is "if I put `assert False` in the first line of `mnist_tutorial_tf.py`, the Travis build should fail"

**Comment by npapernot** [comment]
Ok I see, then we have to make the test run the original tutorial script and check if accuracy is in expected range for both models on legitimate and adversarial inputs

**Comment by paozhuanyinyu-njust** [comment]
Dear @goodfeli, I found the line 182 of utils_tf.py ''acc_value = tf.reduce_mean(keras.metrics.categorical_accuracy(y, model))" should be changed to "acc_value = tf.reduce_mean(tf.cast(keras.metrics.categorical_accuracy(y, model), 'float'))". Thanks for your team and projects.

**Comment by npapernot** [comment]
Closed by #119 #148 #149
